File: NCore/Commands/Commandunban.cs
using NCore;
using NCore.netcraft.server.api;
using System.Threading.Tasks;

namespace NCore
{
    public class Commandunban : Command
    {
        public Commandunban() : base("unban", NCore.GetNCore().lang.get("commands.unban.description"), NCore.GetNCore().lang.get("commands.unban.usage"))
        {
        }

        public override async Task<bool> OnCommand(CommandSender sender, Command cmd, string[] args, string label)
        {
            NCore.Lang lang = sender.IsPlayer ? ((NetcraftPlayer)sender).lang : NCore.GetNCore().lang;
            if (!sender.GetAdmin())
            {
                await sender.SendMessage(lang.get("commands.generic.error.no-perms"));
                return true;
            }

            if (args.Length == 1)
            {
                string a = args[0];
                if(!Netcraft.IsBanned(a))
                {
                    await sender.SendMessage(lang.get("commands.unban.failed.not-banned"));
                    return true;
                }
                await Netcraft.Broadcast(lang.get("commands.unban.success", sender.GetName(), a));
                await Netcraft.UnbanPlayer(a);
            }

            return false;
        }
    }
}

File: NetcraftServer/Commands/Commandunban.vb
Imports NetcraftServer

Public Class Commandunban
    Inherits Command
    Sub New()
        MyBase.New("unban", "Разбанить игрока на сервере", "unban <игрок>")
    End Sub
    Public Overrides Function OnCommand(sender As CommandSender, cmd As Command, args() As String, label As String) As Boolean
        If Not sender.GetAdmin Then
            sender.SendMessage("У Вас недостаточно прав!")
            Return True
        End If
        If args.Length = 1 Then
            Dim a = args(0)
            netcraft.server.api.NCSApi.Netcraft.Broadcast($"{sender.GetName} разбанил игрока {a}.")
            netcraft.server.api.NCSApi.Netcraft.UnbanPlayer(a)
        End If
        Return False
    End Function
End Class
